generate-manuals.js

```diff
@@ -1,152 +1,157 @@
-    lang: 'en',
-    file: 'sanity.md'
+    pages: [
+      {
+        name: 'System documentation for version 9.0',
+        file: 'README.md'
+      },
+      {
+        name: 'Active effects',
+        file: 'effects.md'
+      },
+      {
+        name: 'Actor Importer',
+        file: 'actor_importer.md'
+      },
+      {
+        name: 'Character Creation',
+        file: 'character_creation.md'
+      },
+      {
+        name: 'Chases',
+        file: 'chases.md'
+      },
+      {
+        name: 'Combat',
+        file: 'combat.md'
+      },
+      {
+        name: 'Commands Cheat Sheet',
+        file: 'commands_cheat_sheet.md'
+      },
+      {
+        name: 'Creating your first investigator',
+        file: 'first_investigator.md'
+      },
+      {
+        name: 'Items',
+        file: 'items.md'
+      },
+      {
+        name: 'Item Type: Archetype',
+        file: 'item_archetype.md'
+      },
+      {
+        name: 'Item Type: Book',
+        file: 'item_book.md'
+      },
+      {
+        name: 'Item Type: Occupation',
+        file: 'item_occupation.md'
+      },
+      {
+        name: 'Item Type: Setup',
+        file: 'item_setup.md'
+      },
+      {
+        name: 'Item Type: Skill',
+        file: 'item_skill.md'
+      },
+      {
+        name: 'Links',
+        file: 'links.md'
+      },
+      {
+        name: 'Link creation tool',
+        file: 'link_creation_window.md'
+      },
+      {
+        name: 'Sanity',
+        file: 'sanity.md'
+      }
+    ]
   }
 }
 
 const dbFile = []
-
-for (const id in sources) {
-  try {
-    const md = new Remarkable()
-    const input = fs.readFileSync(
-      './module/manual/' + sources[id].lang + '/' + sources[id].file,
-      'utf8'
-    )
-
-    const source = md.render(input)
-
-    const html = source
-      .replace(/<p>\.<\/p>/g, '<p>&nbsp;</p>')
-      .replace(
-        /\[(fas fa-[^\]]+|game-icon game-icon-[^\]]+)\]/g,
-        '<em class="$1">&nbsp;</em>'
-      )
-      .replace(/src="..\/..\/assets\//g, 'src="systems/CoC7/assets/')
-      .replace(/\n\s*/g, '\n')
-      .replace(/@@coc7./g, '<span>@</span>coc7.')
-
+try {
+  for (const lang in sources) {
+    let id = crypto.createHash('md5').update('manual' + lang).digest('base64').replace(/[\\+=\\/]/g, '').substring(0, 16)
+    while (typeof collisions[id] !== 'undefined') {
+      id = crypto.createHash('md5').update('manual' + lang + Math.random().toString()).digest('base64').replace(/[\\+=\\/]/g, '').substring(0, 16)
+    }
     const dbEntry = {
-      _id: id,
-      name: sources[id].name + ' [' + sources[id].lang + ']',
-      content: '<div class="coc7overview">\n' + html + '\n</div>'
+      name: sources[lang].name + ' [' + lang + ']',
+      pages: [],
+      _id: id
+    }
+    const links = {}
+    for (const source of sources[lang].pages) {
+      let id = crypto.createHash('md5').update('manual' + lang + source.file).digest('base64').replace(/[\\+=\\/]/g, '').substring(0, 16)
```

Comment:
This is called outside of foundry to build manual .db files.
It creates a hash from the key so it is consistent between builds and the converts to base 64 so it uses the same range of characters as FoundryVTTs generator.

Reply:
OK, I see, I forgot the context, now it makes total sense... anyway if I understand it properly then when there is a collision, it adds some random number so I think the consistency between builds will be lost. I imagine this will be a very rare case anyway and I don't think it will have a big impact. 

One suggestion still would be to extract the logic to a function (something like `createReproducibleID`) so with a little refactor lines 87-90 and 97-101 can be just a call to that function (like `createReproducibleID('manual'+lang)`)  and I would include a comment on the function to clarify the use.